diff --git a/batch.c b/batch.c
index a4d19c0..27a23ba 100644
--- a/batch.c
+++ b/batch.c
@@ -22,6 +22,9 @@
 #include "rsync.h"
 #include <zlib.h>
 #include <time.h>
+#ifdef __MVS__
+#include "zos-io.h"
+#endif
 
 extern int eol_nulls;
 extern int recurse;
@@ -252,6 +255,12 @@ void open_batch_files(void)
 		stringjoin(filename, sizeof filename, batch_name, ".sh", NULL);
 
 		batch_sh_fd = open_no_attacker_symlinks(filename, O_WRONLY | O_CREAT | O_TRUNC | O_BINARY, S_IRUSR | S_IWUSR | S_IXUSR);
+		#ifdef __MVS__
+		               if (batch_sh_fd >= 0) {
+		                       __setfdccsid(batch_sh_fd, 819);
+		                       __setfdtext(batch_sh_fd);
+		               }
+		#endif
 		if (batch_sh_fd < 0) {
 			rsyserr(FERROR, errno, "Batch file %s open error", full_fname(filename));
 			exit_cleanup(RERR_FILESELECT);
@@ -294,6 +303,7 @@ void write_batch_shell_file(void)
 	char *p, *p2;
 
 	/* Write argvs info to BATCH.sh file */
+	err |= write(batch_sh_fd, "#!/usr/bin/env bash\n", 20) != 20;
 	err |= write_arg(raw_argv[0]);
 	if (filter_list.head) {
 		if (protocol_version >= 29)
