diff --git a/batch.c b/batch.c
index accc4c6..9a3214d 100644
--- a/batch.c
+++ b/batch.c
@@ -22,6 +22,9 @@
 #include "rsync.h"
 #include <zlib.h>
 #include <time.h>
+#ifdef __MVS__
+#include "zos-io.h"
+#endif
 
 extern int eol_nulls;
 extern int recurse;
@@ -230,6 +233,12 @@ void open_batch_files(void)
 		stringjoin(filename, sizeof filename, batch_name, ".sh", NULL);
 
 		batch_sh_fd = do_open(filename, O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR | S_IXUSR);
+#ifdef __MVS__
+		if (batch_sh_fd >= 0) {
+			__setfdccsid(batch_sh_fd, 819);
+			__setfdtext(batch_sh_fd);
+		}
+#endif
 		if (batch_sh_fd < 0) {
 			rsyserr(FERROR, errno, "Batch file %s open error", full_fname(filename));
 			exit_cleanup(RERR_FILESELECT);
@@ -258,6 +267,7 @@ void write_batch_shell_file(void)
 	char *p, *p2;
 
 	/* Write argvs info to BATCH.sh file */
+	err |= write(batch_sh_fd, "#!/usr/bin/env bash\n", 20) != 20;
 	err |= write_arg(raw_argv[0]);
 	if (filter_list.head) {
 		if (protocol_version >= 29)
