diff --git a/generator.c b/generator.c
index b56fa56..8bd7239 100644
--- a/generator.c
+++ b/generator.c
@@ -543,6 +543,12 @@ void itemize(const char *fnamecmp, struct file_struct *file, int ndx, int statre
 		} else if (preserve_executability
 		 && ((sxp->st.st_mode & 0111 ? 1 : 0) ^ (file->mode & 0111 ? 1 : 0)))
 			iflags |= ITEM_REPORT_PERMS;
+#ifdef __MVS__
+		if (preserve_zos_tags && zos_tags_ndx && S_ISREG(file->mode)) {
+			if (sxp->st.st_tag.ft_ccsid != F_ZOS_CCSID(file) || sxp->st.st_tag.ft_txtflag != F_ZOS_TXTFLAG(file))
+				iflags |= ITEM_REPORT_PERMS;
+		}
+#endif
 		if (uid_ndx && am_root && (uid_t)F_OWNER(file) != sxp->st.st_uid)
 			iflags |= ITEM_REPORT_OWNER;
 		if (gid_ndx && !(file->flags & FLAG_SKIP_GROUP) && sxp->st.st_gid != (gid_t)F_GROUP(file))
@@ -637,8 +643,19 @@ int quick_check_ok(enum filetype ftype, const char *fn, struct file_struct *file
 
 		if (mtime_differs(st, file))
 			return 0;
-		break;
-	  case FT_DIR:
+
+		#ifdef __MVS__
+		if (preserve_zos_tags && zos_tags_ndx) {
+			if (st->st_tag.ft_ccsid != F_ZOS_CCSID(file) || st->st_tag.ft_txtflag != F_ZOS_TXTFLAG(file)) {
+				if (DEBUG_GTE(GENR, 2)) {
+					rprintf(FINFO, "[DEBUG] tags differ: %s (local_ccsid=%u, remote_ccsid=%u, local_txt=%u, remote_txt=%u)\n",
+						fn, st->st_tag.ft_ccsid, F_ZOS_CCSID(file), st->st_tag.ft_txtflag, F_ZOS_TXTFLAG(file));
+				}
+				return 0;
+			}
+		}
+		#endif
+		break;	  case FT_DIR:
 		break;
 	  case FT_SYMLINK: {
 #ifdef SUPPORT_LINKS
@@ -2232,6 +2249,12 @@ void generate_files(int f_out, const char *local_name)
 	int save_info_flist = info_levels[INFO_FLIST];
 	int save_info_progress = info_levels[INFO_PROGRESS];
 
+#ifdef __MVS__
+    if (DEBUG_GTE(GENR, 1)) {
+        rprintf(FINFO, "[DEBUG] generate_files: zos_tags_ndx=%d, preserve_zos_tags=%d\n", zos_tags_ndx, preserve_zos_tags);
+    }
+#endif
+
 	if (protocol_version >= 29) {
 		itemizing = 1;
 		maybe_ATTRS_REPORT = stdout_format_has_i ? 0 : ATTRS_REPORT;
