diff --git a/testsuite/hardlinks.test b/testsuite/hardlinks.test
index c02db3f..e3f214a 100644
--- a/testsuite/hardlinks.test
+++ b/testsuite/hardlinks.test
@@ -11,6 +11,9 @@
 
 . "$suitedir/rsync.fns"
 
+# Add --no-zos-tags to avoid 'p' in itemized output on z/OS
+RSYNC="$RSYNC --no-zos-tags"
+
 SSH="$scratchdir/src/support/lsh.sh"
 
 # Build some hardlinks
@@ -82,7 +85,7 @@ diff $diffopt "$name1" "$todir" || test_fail "solo copy of name1 failed"
 rm -rf "$fromdir" "$todir"
 makepath "$fromdir/sym" "$todir"
 $RSYNC -aH "$fromdir/sym" "$todir"
-diff $diffopt "$fromdir" "$todir" || test_fail "solo copy of sym failed"
+diff $diffopt "$fromdir/sym" "$todir/sym" || test_fail "solo copy of sym failed"
 
 # The script would have aborted on error, so getting here means we've won.
 exit 0
