diff --git a/configure.ac b/configure.ac
index b6de4eb3..8d6643ad 100644
--- a/configure.ac
+++ b/configure.ac
@@ -131,11 +131,6 @@ fi
 # I don't want our version too far out of sync.
 CFLAGS="$CFLAGS -DHAVE_CONFIG_H"
 
-# If GCC, turn on warnings.
-if test x"$GCC" = x"yes"; then
-	CFLAGS="$CFLAGS -Wall -W"
-fi
-
 AC_ARG_WITH(rrsync,
         AS_HELP_STRING([--with-rrsync],[also install the rrsync script and its manpage]))
 if test x"$with_rrsync" != x"yes"; then
