diff --git a/rsync.h b/rsync.h
index e29c37c3..70fa6e44 100644
--- a/rsync.h
+++ b/rsync.h
@@ -488,6 +488,13 @@ enum delret {
 #include <sys/sysmacros.h>
 #endif
 
+#ifdef __MVS__
+  /* z/OS has no 'makedev' so create trivial services */
+  #define makedev(devmajor,devminor) ((dev_t)((devmajor) << 24 | (devminor)))
+  #define minor(dev)                 ((dev) & 0xFFFFFF)
+  #define major(dev)                 (((uint32)(dev)) >> 24)
+#endif
+
 #ifdef MAKEDEV_TAKES_3_ARGS
 #define MAKEDEV(devmajor,devminor) makedev(0,devmajor,devminor)
 #else
@@ -1274,11 +1281,19 @@ extern int errno;
 #define CHMOD_BITS (S_ISUID | S_ISGID | S_ISVTX | ACCESSPERMS)
 
 #ifndef _S_IFMT
-#define _S_IFMT        0170000
+  #ifdef S_IFMT
+    #define _S_FMT S_IFMT
+  #else
+    #define _S_IFMT        0170000
+  #endif
 #endif
 
 #ifndef _S_IFLNK
-#define _S_IFLNK  0120000
+  #ifdef S_IFLNK
+    #define _S_IFLNK S_IFLNK
+  #else
+    #define _S_IFLNK  0120000
+  #endif
 #endif
 
 #ifndef S_ISLNK
