diff --git a/rsync.h b/rsync.h
index 2c5e5376..6f9d74f6 100644
--- a/rsync.h
+++ b/rsync.h
@@ -449,6 +449,13 @@ enum delret {
 #endif
 #endif
 
+#ifdef HAVE_SYS_MODES_H
+/* apparently z/OS needs this for S_ISLNK */
+#ifndef S_ISLNK
+#include <sys/modes.h>
+#endif
+#endif
+
 /* these are needed for the uid/gid mapping code */
 #include <pwd.h>
 #include <grp.h>
@@ -488,6 +495,13 @@ enum delret {
 #include <sys/sysmacros.h>
 #endif
 
+#ifdef __MVS__
+  /* z/OS has no 'makedev' so create trivial services */
+  #define makedev(devmajor,devminor) ((dev_t)((devmajor) << 24 | (devminor)))
+  #define minor(dev)                 ((dev) & 0xFFFFFF)
+  #define major(dev)                 (((uint32)(dev)) >> 24)
+#endif
+
 #ifdef MAKEDEV_TAKES_3_ARGS
 #define MAKEDEV(devmajor,devminor) makedev(0,devmajor,devminor)
 #else
