diff --git a/rsync.h b/rsync.h
index 479ac48..420a637 100644
--- a/rsync.h
+++ b/rsync.h
@@ -317,6 +317,8 @@ enum delret {
 #include "config.h"
 
 /* The default RSYNC_RSH is always set in config.h. */
+# include <sys/stat.h>
+#include <dirent.h>
 
 #include <stdio.h>
 #ifdef HAVE_SYS_TYPES_H
@@ -444,6 +446,13 @@ enum delret {
 #endif
 #endif
 
+#ifdef HAVE_SYS_MODES_H
+/* apparently z/OS needs this for S_ISLNK */
+#ifndef S_ISLNK
+#include <sys/modes.h>
+#endif
+#endif
+
 /* these are needed for the uid/gid mapping code */
 #include <pwd.h>
 #include <grp.h>
@@ -459,6 +468,8 @@ enum delret {
 #include <sys/file.h>
 #endif
 
+#define HAVE_DIRENT_H 1
+
 #ifdef HAVE_DIRENT_H
 # include <dirent.h>
 #else
@@ -483,6 +494,13 @@ enum delret {
 #include <sys/sysmacros.h>
 #endif
 
+#ifdef __MVS__
+  /* z/OS has no 'makedev' so create trivial services */
+  #define makedev(devmajor,devminor) ((dev_t)((devmajor) << 24 | (devminor)))
+  #define minor(dev)                 ((dev) & 0xFFFFFF)
+  #define major(dev)                 (((uint32)(dev)) >> 24)
+#endif
+
 #ifdef MAKEDEV_TAKES_3_ARGS
 #define MAKEDEV(devmajor,devminor) makedev(0,devmajor,devminor)
 #else
