diff --git a/delete.c b/delete.c
index 4a294853..cb930580 100644
--- a/delete.c
+++ b/delete.c
@@ -127,7 +127,7 @@ static enum delret delete_dir_contents(char *fname, uint16 flags)
  * Note that fbuf must point to a MAXPATHLEN buffer if the mode indicates it's
  * a directory! (The buffer is used for recursion, but returned unchanged.)
  */
-enum delret delete_item(char *fbuf, uint16 mode, uint16 flags)
+enum delret delete_item(char *fbuf, uint32 mode, uint16 flags)
 {
 	enum delret ret;
 	char *what;
@@ -224,7 +224,7 @@ enum delret delete_item(char *fbuf, uint16 mode, uint16 flags)
 	return ret;
 }
 
-uint16 get_del_for_flag(uint16 mode)
+uint16 get_del_for_flag(uint32 mode)
 {
 	if (S_ISREG(mode))
 		return DEL_FOR_FILE;
